JSII compilation error related to @types/jest

Who else receives this error on a fresh initialized construct-library project using projen?
```
09:01:37 - Starting compilation in watch mode...

[2022-06-18T09:01:40.722] [ERROR] jsii/compiler - Compilation errors prevented the JSII assembly from being created
node_modules/@sinclair/typebox/typebox.d.ts(237,21): error TS2456: Type alias 'StaticIntersectReduce' circularly references itself.
node_modules/@sinclair/typebox/typebox.d.ts(237,110): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(237,124): error TS2315: Type 'StaticIntersectReduce' is not generic.
node_modules/@sinclair/typebox/typebox.d.ts(256,69): error TS2315: Type 'StaticIntersectReduce' is not generic.
node_modules/@sinclair/typebox/typebox.d.ts(271,105): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(272,98): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(285,40): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(289,44): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(291,40): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(323,128): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(325,128): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(329,64): error TS2574: A rest element type must be an array type.
node_modules/@sinclair/typebox/typebox.d.ts(331,61): error TS2574: A rest element type must be an array type.
node_modules/@types/prettier/index.d.ts(41,54): error TS2315: Type 'IsTuple' is not generic.
node_modules/@types/prettier/index.d.ts(53,6): error TS2456: Type alias 'IsTuple' circularly references itself.
node_modules/@types/prettier/index.d.ts(53,65): error TS2574: A rest element type must be an array type.
node_modules/@types/prettier/index.d.ts(53,84): error TS2315: Type 'IsTuple' is not generic.
node_modules/@types/prettier/index.d.ts(96,5): error TS2589: Type instantiation is excessively deep and possibly infinite.
node_modules/@types/prettier/index.d.ts(131,5): error TS2589: Type instantiation is excessively deep and possibly infinite.
node_modules/@types/prettier/index.d.ts(165,5): error TS2589: Type instantiation is excessively deep and possibly infinite.

09:01:40 - Found 20 errors. Watching for file changes.
```

It seems to be somehow related to the used version of `"@types/jest": "^28.1.2"` .
Tried to downgrade to 27.4.1  but adding `devDeps: ["@types/jest@27.4.1"]  ` to my projen config dir not work. In the resulting package.json there is still version 28.1.2 added.
But using `peerDeps: ["@types/jest@27.4.1"]`  did the trick. And the right version is set in the package.json.

Now I am getting the following error:
```
❯ pj watch
👾 watch | jsii -w --silence-warnings=reserved-word
[2022-06-18T11:19:25.509] [ERROR] jsii/package-info - Unable to find a JSII dependency named "@types/jest" as "27.4.1". If you meant to include a non-JSII dependency, try adding it to bundledDependencies instead.
Error: Error: Unable to locate jsii assembly for "@types/jest". If this module is not jsii-enabled, it must also be declared under bundledDependencies: Error [ERR_PACKAGE_PATH_NOT_EXPORTED]:
```

The only way to make compilation work is, to add `  bundledDeps: ['@types/jest@27.4.1']` to my projen config.

--- a/.projen/tasks.json
+++ b/.projen/tasks.json
@@ -392,7 +392,7 @@
           "exec": "yarn install --check-files"
         },
         {
-          "exec": "yarn upgrade @types/conventional-changelog-config-spec @types/fs-extra @types/glob @types/ini @types/jest @types/node @types/semver @types/yargs @typescript-eslint/eslint-plugin @typescript-eslint/parser all-contributors-cli esbuild eslint-config-prettier eslint-import-resolver-node eslint-import-resolver-typescript eslint-plugin-import eslint-plugin-prettier eslint jest-junit jest jsii jsii-diff jsii-docgen jsii-pacmak json-schema json5 npm-check-updates prettier standard-version ts-jest typescript @iarna/toml case chalk conventional-changelog-config-spec fs-extra glob ini semver shx xmlbuilder2 yaml yargs"
+          "exec": "yarn upgrade @types/conventional-changelog-config-spec @types/fs-extra @types/glob @types/ini @types/jest @types/node @types/semver @types/yargs @typescript-eslint/eslint-plugin @typescript-eslint/parser all-contributors-cli esbuild eslint-config-prettier eslint-import-resolver-node eslint-import-resolver-typescript eslint-plugin-import eslint-plugin-prettier eslint jest jest-junit jsii jsii-diff jsii-docgen jsii-pacmak json-schema json5 npm-check-updates prettier standard-version ts-jest typescript @iarna/toml case chalk conventional-changelog-config-spec fs-extra glob ini semver shx xmlbuilder2 yaml yargs"
         },
         {
           "exec": "/bin/bash ./projen.bash"

--- a/docs/api/API.md
+++ b/docs/api/API.md
@@ -7387,6 +7387,7 @@ new javascript.Jest(project: NodeProject, options?: JestOptions)
 Name | Type | Description 
 -----|------|-------------
 **config**🔹 | <code>any</code> | Escape hatch.
+**jestVersion**🔹 | <code>string</code> | <span></span>
 
 ### Methods
 

--- a/package.json
+++ b/package.json
@@ -52,7 +52,7 @@
     "@types/fs-extra": "^8",
     "@types/glob": "^7.2.0",
     "@types/ini": "^1.3.31",
-    "@types/jest": "^27.5.2",
+    "@types/jest": "^27",
     "@types/node": "^14",
     "@types/semver": "^7.3.10",
     "@types/yargs": "^15.0.14",

--- a/src/javascript/jest.ts
+++ b/src/javascript/jest.ts
@@ -519,6 +519,8 @@ export interface JestOptions {
   /**
    * The version of jest to use.
    *
+   * Note that same version is used as version of `@types/jest` and `ts-jest` (if Typescript in use), so given version should work also for those.
+   *
    * @default - installs the latest jest version
    */
   readonly jestVersion?: string;
@@ -567,6 +569,7 @@ export class Jest {
    * Escape hatch.
    */
   public readonly config: any;
+  public readonly jestVersion: string;
 
   private readonly testMatch: string[];
   private readonly ignorePatterns: string[];
@@ -590,11 +593,8 @@ export class Jest {
 
     // Jest snapshot files are generated files!
     project.root.annotateGenerated("*.snap");
-
-    const jestDep = options.jestVersion
-      ? `jest@${options.jestVersion}`
-      : "jest@^27"; // pinning at version 27 for now because of an issue: https://github.com/projen/projen/issues/1801
-    project.addDevDeps(jestDep);
+    this.jestVersion = options.jestVersion ? `@${options.jestVersion}` : "";
+    project.addDevDeps(`jest${this.jestVersion}`);
 
     this.jestConfig = options.jestConfig;
 

--- a/src/typescript/typescript.ts
+++ b/src/typescript/typescript.ts
@@ -379,7 +379,7 @@ export class TypeScriptProject extends NodeProject {
    * for us. just run them directly from javascript.
    */
   private addJestCompiled(jest: Jest) {
-    this.addDevDeps("@types/jest");
+    this.addDevDeps(`@types/jest${jest.jestVersion}`);
 
     const testout = path.posix.relative(this.srcdir, this.testdir);
     const libtest = path.posix.join(this.libdir, testout);
@@ -428,7 +428,10 @@ export class TypeScriptProject extends NodeProject {
   }
 
   private addJestNoCompile(jest: Jest) {
-    this.addDevDeps("@types/jest", "ts-jest@^27"); // pinning for now because of an issue: https://github.com/projen/projen/issues/1813
+    this.addDevDeps(
+      `@types/jest${jest.jestVersion}`,
+      `ts-jest${jest.jestVersion}`
+    );
 
     jest.addTestMatch(`<rootDir>/${this.srcdir}/**/__tests__/**/*.ts?(x)`);
     jest.addTestMatch(

--- a/test/javascript/node-project.test.ts
+++ b/test/javascript/node-project.test.ts
@@ -678,7 +678,6 @@ test("enabling renovatebot does not overturn mergify: false", () => {
   expect(snapshot).toHaveProperty(["renovate.json5"]);
   expect(json5.parse(snapshot["renovate.json5"]).ignoreDeps).toEqual([
     "jest-junit",
-    "jest",
     "npm-check-updates",
     "standard-version",
     "projen",

--- a/test/javascript/upgrade-dependencies.test.ts
+++ b/test/javascript/upgrade-dependencies.test.ts
@@ -34,7 +34,7 @@ test("upgrades command doesn't include ignored packages", () => {
     },
   });
 
-  const deps = "jest-junit jest npm-check-updates projen standard-version dep1";
+  const deps = "jest jest-junit npm-check-updates projen standard-version dep1";
 
   const tasks = synthSnapshot(project)[TaskRuntime.MANIFEST_FILE].tasks;
   expect(tasks.upgrade.steps[7].exec).toStrictEqual(`yarn upgrade ${deps}`);

